accord-core/src/main/java/accord/topology/TopologyManager.java

```diff
@@ -892,8 +1055,11 @@ private <C, K extends Routables<?>, T> T atLeast(K select, long minEpoch, long m
 
         if (i == snapshot.epochs.length)
         {
-            if (!select.isEmpty())
-                throw new IllegalArgumentException("Ranges " + select + " could not be found");
+            // Epochs earlier than minEpoch might have been GC'd, so we can not collect
+            // matching ranges for them. However, if ranges were still present in the min epoch,
+            // we have reported them.
+            if (!select.isEmpty() && !select.without(snapshot.get(minEpoch).global.ranges).isEmpty())
```

Comment:
maybe move to `Invariants.requireArgument`, or creating the exception with `Invariants.illegalArgument`?